testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -553,7 +553,11 @@
 
 Then(/^I should be logged in$/) do
   xpath_query = "//a[@href='/rhn/Logout.do']"
-  raise 'User is not logged in' unless find(:xpath, xpath_query)
+  if has_selector?(:xpath, xpath_query)
+    # The element is present
+  else
+    raise 'User is not logged in' unless find(:xpath, xpath_query)
+  end
```

Comment:
I would instead append wait: X here.
As I proposed yesterday, this will setup a specific timeout, and meanwhile the find method will refresh the DOM until it find the element or raise the timeout.